com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StartWithoutDebuggingRequestsHandler.java

```diff
@@ -0,0 +1,168 @@
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package com.microsoft.java.debug.core.adapter.handler;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+import com.google.gson.JsonObject;
+import com.microsoft.java.debug.core.DebugException;
+import com.microsoft.java.debug.core.adapter.AdapterUtils;
+import com.microsoft.java.debug.core.adapter.ErrorCode;
+import com.microsoft.java.debug.core.adapter.IDebugAdapterContext;
+import com.microsoft.java.debug.core.adapter.LaunchMode;
+import com.microsoft.java.debug.core.protocol.Events;
+import com.microsoft.java.debug.core.protocol.JsonUtils;
+import com.microsoft.java.debug.core.protocol.Messages.Request;
+import com.microsoft.java.debug.core.protocol.Messages.Response;
+import com.microsoft.java.debug.core.protocol.Requests.Arguments;
+import com.microsoft.java.debug.core.protocol.Requests.CONSOLE;
+import com.microsoft.java.debug.core.protocol.Requests.Command;
+import com.microsoft.java.debug.core.protocol.Requests.DisconnectArguments;
+import com.microsoft.java.debug.core.protocol.Requests.LaunchArguments;
+import com.microsoft.java.debug.core.protocol.Requests.RunInTerminalRequestArguments;
+import com.sun.jdi.connect.IllegalConnectorArgumentsException;
+import com.sun.jdi.connect.VMStartException;
+
+public class StartWithoutDebuggingRequestsHandler extends AbstractLaunchRequestHandler {
+    private static final String TERMINAL_TITLE = "Java Process Console";
+    private Process debuggeeProcess;
+
+    @Override
+    public List<Command> getTargetCommands() {
+        return Arrays.asList(Command.values());
+    }
+
+    @Override
+    public CompletableFuture<Response> handle(Command command, Arguments arguments, Response response,
+            IDebugAdapterContext context) {
+        if (Command.INITIALIZE.equals(command)) {
+            return CompletableFuture.completedFuture(response);
+        } else if (Command.LAUNCH.equals(command)) {
+            LaunchArguments launchArguments = (LaunchArguments) arguments;
+            return launchArguments.noDebug ? handleLaunchCommand(arguments, response, context) : CompletableFuture.completedFuture(response);
+        } else if (Command.DISCONNECT.equals(command)) {
+            return handleDisconnectCommand(arguments, response, context);
+        } else {
+            if (context.getLaunchMode() == LaunchMode.NO_DEBUG) {
+                String errorMsg = String.format("Unrecognized request: { _request: %s }", command);
+                throw AdapterUtils.createCompletionException(errorMsg, ErrorCode.UNRECOGNIZED_REQUEST_FAILURE);
+            }
+            return CompletableFuture.completedFuture(response);
+        }
```

Comment:
Add a method to detect if the launchMode is noDebug or not.
```
private boolean isNoDebug(Command command, Arguments arguments, IDebugAdapterContext context) {
    return context.getLaunchMode() == LaunchMode.NO_DEBUG || (Command.LAUNCH == command && ((LaunchArgument) arguments).noDebug);
}
```

Then the `handle` logic could be 
```
if (isNoDebug()) {
    if (launch) ...
    else if (disconnect) ...
    else unrecognized request...
}
do nothing and return response directly...
```